config/initializers/message_bus.rb

```diff
@@ -1,18 +1,29 @@
 # frozen_string_literal: true
 
-if Rails.env.test?
-  MessageBus.configure(backend: :memory)
-  MessageBus.keepalive_interval = 0
-  MessageBus.off
-else
-  MessageBus.redis_config = System::Application.config.redis.merge(db: 8)
-  MessageBus.timer.on_error do |error|
-    System::ErrorReporting.report_error(error)
+Class.new do
```

Comment:
You create an anonymous class, meaning that it might be different for multi threaded application
Any reason for that?

Reply:
Initializers are called only once.